src/Microsoft.Windows.ComputeVirtualization/Process.cs

```diff
@@ -67,45 +75,34 @@ internal Process(string id, int pid, StreamWriter stdin, StreamReader stdout, St
         /// <param name="width">The new width, in character cells.</param>
         public void ResizeConsole(short height, short width)
         {
-            HcsFunctions.ResizeConsoleInComputeSystem(_id, _pid, height, width, 0);
-        }
+            Schema.ProcessConsoleSize procSize = new Schema.ProcessConsoleSize();
+            procSize.Height = (ushort)height;
+            procSize.Width = (ushort)width;
 
-        /// <summary>
-        /// Waits forever for the process to exit.
-        /// </summary>
-        public void WaitForExit()
-        {
-            int exitCode;
-            int hr = HcsFunctions.WaitForProcessInComputeSystem(_id, _pid, -1, out exitCode);
-            if (hr < 0)
-            {
-                throw new Win32Exception(hr);
-            }
-            _exitCode = exitCode;
-            _exited = true;
+            Schema.ProcessModifyRequest procModReq = new Schema.ProcessModifyRequest();
+            procModReq.Operation = Schema.ProcessModifyOperation.ConsoleSize;
+            procModReq.ConsoleSize = procSize;
+
+            string result;
+            HcsFunctions.ProcessHcsCall(HcsFunctions.HcsModifyProcess(_p, JsonHelper.ToJson(procModReq), out result));
         }
 
         /// <summary>
         /// Waits for the process to exit.
         /// </summary>
-        /// <param name="timeout">The number of milliseconds to wait.</param>
+        /// <param name="timeout">The number of milliseconds to wait. Default is infinite</param>
         /// <returns>True if the process exited, false if the wait timed out.</returns>
-        public bool WaitForExit(int timeout)
+        public bool WaitForExit(int timeout = Timeout.Infinite)
```

Comment:
Can we use Timespans instead. There are more conversion bugs than anything. WaitForExit(TimeSpan.FromSeconds(30)). "Not eh. What is timeout? MS oh, x * 1000"